### PR TITLE
[Feat][C++] Add isValid for property in high level

### PR DESCRIPTION
**Pull request by jasinliu**
## Proposed changes

Related to #434

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/alibaba/GraphAr/blob/main/CONTRIBUTING.rst) doc
- [x] I have signed the CLA
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

If it works, i will add test case for it.

### CODE REVIEW

**Comment by jasinliu** [comment]
The method and the result turn to this format.
image
image

File: cpp/include/gar/graph.h [hunk 1]
```diff
@@ -75,6 +75,19 @@ class Vertex {
...
+   * @brief Return true if value at the property is valid (not null).
+   *
+   * @param property The property name.
+   * @return True if value at the property is valid, False otherwise.
+   */
+  inline bool IsValid(const std::string& property) const {
+    if (properties_.find(property) != properties_.end()) {
+      return properties_.at(property).has_value();
+    }
+    return true;
```
Comment (lixueclaire) [inline cpp/include/gar/graph.h:88]: I'm confused as to why the function returns true when the name is not found?

Comment (jasinliu) [reply cpp/include/gar/graph.h:88]: Sorry, this is a very simple implementation. It doesn't validate the list_properties_. I wonder whether the return value should be true when the length of the list value is 0?

Comment (lixueclaire) [reply cpp/include/gar/graph.h:88]: > Sorry, this is a very simple implementation. It doesn't validate the list_properties_. I wonder whether the return value should be true when the length of the list value is 0?

Given that Arrow employs a distinct field to indicate null values, I think an empty array should not be equated with null.

Comment (jasinliu) [reply cpp/include/gar/graph.h:88]: OK, I can resolve it.

**Review by lixueclaire — APPROVED** [review]
LGTM now, thanks for you contribution!

**Comment by acezen** [comment]
Hi, @jasinliu, Can you add this `IsValid` to Edge too? I think they both need the API for property value check

**Comment by jasinliu** [comment]
> Hi, @jasinliu, Can you add this `IsValid` to Edge too? I think they both need the API for property value check

Sure! I think `IsValid` is added to Edge property. I will add some error handlers, docs and test cases.

**Comment by acezen** [comment]
> > Hi, @jasinliu, Can you add this `IsValid` to Edge too? I think they both need the API for property value check
> 
> Sure! I think `IsValid` is added to Edge property. I will add some error handlers, docs and test cases.
 That would be great. When you think the PR is ready, feel free to ping me or @lixueclaire to merged it.

**Comment by jasinliu** [comment]
I need a review due to that the test runs fine on my ubuntu machine, but fails on the [action](https://github.com/GraphScope/GraphAr/actions/runs/8521177757/job/23338817844) @acezen @lixueclaire

**Comment by lixueclaire** [comment]
> I need a review due to that the test runs fine on my ubuntu machine, but fails on the [action](https://github.com/GraphScope/GraphAr/actions/runs/8521177757/job/23338817844) @acezen @lixueclaire

hi, @jasinliu Without detailed error logs, it's challenging to diagnose the exact problem. Could you try incrementally commenting out sections of the test to isolate and identify the failing component?

**Comment by jasinliu** [comment]
> hi, @jasinliu Without detailed error logs, it's challenging to diagnose the exact problem. Could you try incrementally commenting out sections of the test to isolate and identify the failing component?

OK. Let me try

**Comment by jasinliu** [comment]
> > hi, @jasinliu Without detailed error logs, it's challenging to diagnose the exact problem. Could you try incrementally commenting out sections of the test to isolate and identify the failing component?
> 
> OK. Let me try

The problem is the data in gar-test with absolute path. I want to remove it.[Here](https://github.com/GraphScope/gar-test/pull/16)

**Comment by acezen** [comment]
> > > hi, @jasinliu Without detailed error logs, it's challenging to diagnose the exact problem. Could you try incrementally commenting out sections of the test to isolate and identify the failing component?
> > 
> > 
> > OK. Let me try
> 
> The problem is the data in gar-test with absolute path. I want to remove it.[Here](https://github.com/GraphScope/gar-test/pull/16)

The [testing update PR](https://github.com/GraphScope/gar-test/pull/16) has been merged, you can update the testing submodule and run the CI again.

**Comment by acezen** [comment]
@jasinliu you may need to checkout the testing to the latest commit to apply your change in testing repo.

**Review by acezen — APPROVED** [review]
LGTM

**Pull request merged**